libraries/ts-command-line/src/providers/ScopedCommandLineAction.ts

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { CommandLineAction, ICommandLineActionOptions } from './CommandLineAction';
+import { CommandLineParser, ICommandLineParserOptions } from './CommandLineParser';
+import { CommandLineParserExitError } from './CommandLineParserExitError';
+import type { CommandLineParameter } from '../parameters/BaseClasses';
+import type { CommandLineParameterProvider, ICommandLineParserData } from './CommandLineParameterProvider';
+
+interface IInternalScopedCommandLineParserOptions extends ICommandLineParserOptions {
+  readonly actionOptions: ICommandLineActionOptions;
+  readonly unscopedActionParameters: ReadonlyArray<CommandLineParameter>;
+  readonly onDefineScopedParameters: (commandLineParameterProvider: CommandLineParameterProvider) => void;
+  readonly onExecute: () => Promise<void>;
+}
+
+class InternalScopedCommandLineParser extends CommandLineParser {
```

Comment:
Can you leave a comment here mentioning that this is generated after the first pass of the parser and when it's executed?